### DISCUSSION TITLE
different languages support

### DISCUSSION THREAD

**Discussion by Newdea** [snezhig/obsidian-front-matter-title#157] [General]
This is a great plugin, have you considered supporting different languages?

**Comment by snezhig** [comment]
Thank you.
Do you mean descriptions of features and docs? 
Which languages?

**Reply by Newdea** [reply]
just interface descriptions ,Just localize the user interface, e.g. [obsidian-spaced-repetition](https://github.com/st3v3nmw/obsidian-spaced-repetition) plugin, the settings interface, command input (ctrl+p) can be displayed as local language

As for the documentation tutorials, you don't have to deal with them on purpose, a basic web page translation will do.
Of course, I'm planning to write a post about this plugin in the near future to introduce it to my local friends, this plugin is just too convenient for those who writing notes in the ZK way!

![image](https://github.com/snezhig/obsidian-front-matter-title/assets/9208450/1e65d3a0-f81d-40d7-891e-1b0f0dadb7b8)


![image](https://github.com/snezhig/obsidian-front-matter-title/assets/9208450/bd651b73-fd85-4634-a6b1-a7a130f34184)

**Reply by snezhig** [reply]
I know two languages only, but I can use google translate. I will try to add mechanism to change language.

**Reply by Newdea** [reply]
thks a lot. 
I can translate the Chinese part if needed.

**Reply by snezhig** [reply]
Base logic in progress.
![изображение](https://github.com/snezhig/obsidian-front-matter-title/assets/3987716/f3d7a632-c2ec-4c84-a872-cae604791e24)

**Comment by snezhig** [comment]
Here is English translation: https://github.com/snezhig/obsidian-front-matter-title/blob/master/src/i18n/en/index.ts
Folders with empty files are already created: https://github.com/snezhig/obsidian-front-matter-title/tree/master/src/i18n

Also feel free to suggest any changes for translation structure.

**Comment by Newdea** [comment]
Hello, I've translated it as Chinese and it looks amazing.
But there is also a part of the display in English, did you  forgot to add it, it is recommended to add it to the i18n.

![image](https://github.com/snezhig/obsidian-front-matter-title/assets/9208450/3628792c-99aa-42eb-b6f0-87890c3b3a05)

---

![image](https://github.com/snezhig/obsidian-front-matter-title/assets/9208450/76447871-75e3-4a6c-b2f6-f07dcc41b1bb)

![image](https://github.com/snezhig/obsidian-front-matter-title/assets/9208450/1ec152ef-9f13-4dc6-97bf-aa91a0c66590)

![image](https://github.com/snezhig/obsidian-front-matter-title/assets/9208450/b6b1eb9d-c8e4-4003-b2ea-07fd8c129e60)

**Reply by snezhig** [reply]
I forgot something, yes.

**Reply by snezhig** [reply]
Pull dev branch to get new translations.
Changes: https://github.com/snezhig/obsidian-front-matter-title/pull/161/files

**Reply by Newdea** [reply]
ok, i update it.

https://github.com/snezhig/obsidian-front-matter-title/pull/162/files
